### ISSUE TITLE
Outline style to be configurable.

### ISSUE THREAD

**Issue by alan-cole** [civictheme/uikit#428]
### User story

As a theming developer
I want define the style of outline used for focus indicators
So that I can match established branding for that website

### Description

While `outline width` and `outline offset` are configurable variables, the `outline-style` is not.

Further to this, the outline-style used in input fields is not using the `ct-outline()` mixin, but is re-defined.

This means that updating the style from `solid` to anything else requires multiple overrides.

There should be a `$ct-outline-style: solid !default;` variable defined in variables.components.scss, and used throughout for focus indicators.

### Acceptance criteria

Given I am a theming deveoper
When I add `$ct-outline-style: dotted;` in variables.components.scss
Then I should see all focus indicators use the dotted style.

**Comment by alan-cole** [comment]
Further to this,

1. The ct-button doesn't define the `outline-width: $ct-outline-width;` on it's focus indicator.
2. `.ct-menu__item__link-trigger` doesn't set a `ct-outline()`
3. `.ct-logo` doesn't set a `ct-outline()`
3. `.ct-tag` doesn't set a `ct-outline()`
5. `.ct-back-to-top__button` should not have an outline-color: transparent. It should use `$ct-outline-light`.
